src/aihwkit/simulator/tiles/module.py

```diff
@@ -227,23 +224,31 @@ def to(self, *args: Any, **kwargs: Any) -> "TileModule":
         if rpu_config is not None:
             self.replace_with(rpu_config)
 
-        device = None
-        if "device" in kwargs:
-            device = kwargs["device"]
-        elif len(new_args) > 0 and not isinstance(new_args[0], (Tensor, dtype)):
-            if new_args[0] is None:
-                return self
-            device = torch_device(args[0])
+        device = kwargs.get("device")
+        dtype = kwargs.get("dtype")
+        for arg in args:
+            if isinstance(arg, bool):
+                continue
+            if isinstance(arg, torch_dtype):
+                dtype = arg
+            if isinstance(arg, (str, torch_device)):
+                device = torch_device(arg) if isinstance(arg, str) else arg
 
```

Comment:
What was the problem here? Not sure whether the suggested solution is better than the original version. Does kwargs.get("device") does not throw an error when it is not found for instance?

Reply:
Yes, the `.get` was wrong. Don't know why this wasn't picked up by the tests.
I think the new logic handles all the cases correctly when we pass the `device` and `dtype` as either args and kwargs. The old logic didn't handle that correctly.